app/src/main/kotlin/com/simplemobiletools/gallery/pro/adapters/MediaAdapter.kt

```diff
@@ -234,11 +234,19 @@ class MediaAdapter(
     }
 
     private fun renameFile() {
+        val firstPath = getFirstSelectedItemPath() ?: return
+
+        val isSDOrOtgRootFolder = activity.isAStorageRootFolder(firstPath.getParentPath()) && !firstPath.startsWith(activity.internalStoragePath)
+        if (isRPlus() && isSDOrOtgRootFolder) {
+            activity.toast(R.string.rename_in_sd_card_system_restriction)
```

Comment:
use Toast.LENGTH_LONG for such longer error message please, so there is enough time for reading them